WIP Show itinerary results

akwc10 opened the pull request.

akwc10 committed:
WIP Itinerary results activity

akwc10 committed:
WIP Pass result from repo to MainActivity to ItineraryResultActivity

akwc10 committed:
WIP No idea how to parse TO to domain model without breaking clean architecture

akwc10 committed:
WIP First attempt to present domain model always returns null. Help!!!

akwc10 committed:
Rest of commit

akwc10 committed:
Removed results serialization

fsbarata commented on api/src/main/java/com/my/api/JsonOperations.kt:
```diff
@@ -0,0 +1,16 @@
+    if (this.has(TYPE)) {
+        if (this.getString(TYPE).contains(resultType)) return true
+    }
+    return false
```
can be simplified

fsbarata commented on api/src/main/java/com/my/api/IRetrofitApi.kt:
```diff
@@ -12,4 +12,6 @@ interface IJourneyPlannerRepository {
 
     fun cancelAsyncCall(call: Call<JourneyPlannerResult>?)
+
+    fun getJourneyPlannerDomainModel(): JourneyPlannerResultDomainModel?
```
no, this should be supplied in the callback (when you enqueue)

fsbarata commented on api/src/main/java/com/my/api/IRetrofitApi.kt:
```diff
@@ -1,9 +1,9 @@
 import retrofit2.Call
 
-interface IJourneyPlannerRepository {
+interface IRetrofitApi {
```
this cannot be called IRetrofitApi. It has to be specific to the journey planner

fsbarata commented on api/src/main/java/com/my/api/RetrofitApi.kt:
```diff
@@ -1,15 +1,17 @@
+private val logger = KotlinLogging.logger {}
+
+class RetrofitApi : IRetrofitApi {
+    var journeyPlannerResultDomainModel: JourneyPlannerResultDomainModel? = null
```
do not use state. Everything should be supplied and consumed as it exists

fsbarata commented on api/src/main/java/com/my/api/RetrofitApi.kt:
```diff
@@ -35,13 +36,20 @@ class JourneyPlannerRepository : IJourneyPlannerRepository {
-                    Log.d(TAG, "body(): ${response.body()}")
-//                TODO("Display data in new activity")
+                    val responseBodyString = response.body().toString()
+                    journeyPlannerResultDomainModel = when {
```
instead of recording it, you need to call the callback

fsbarata commented on api/src/main/java/com/my/api/TransformOperations.kt:
```diff
@@ -0,0 +1,46 @@
+            mutableListOf<JourneyPlannerResultDomainModel.Itinerary.Journey.Leg>()
+
+        journey.legs.forEach {
+            legs.add(
```
this is a map operation

val legs = journey.legs.map {
     line below
}

fsbarata commented on api/src/main/java/com/my/api/TransformOperations.kt:
```diff
@@ -0,0 +1,46 @@
+            )
+        }
+
+        journeys.add(
```
same as mentioned, this is a map operation

fsbarata commented on app/src/main/AndroidManifest.xml:
```diff
@@ -4,12 +4,17 @@
+        <activity
+            android:name=".views.itineraryresults.ItineraryResultsActivity"
+            android:parentActivityName=".MainActivity"
+            android:screenOrientation="portrait"></activity>
```
can be simplified by using /> instead of ></activity>

fsbarata commented on app/src/main/java/com/my/journeyplanner/MainActivity.kt:
```diff
@@ -1,35 +1,32 @@
-    )
 
-    private val mainPresenter by lazy { MainPresenter(this, interactors) }
+    private val mainPresenter by lazy { MainPresenter(this, RetrofitApi()) }
```
I will teach you dependency injection soon

fsbarata commented on app/src/main/java/com/my/journeyplanner/presenters/main/MainPresenter.kt:
```diff
@@ -1,11 +1,11 @@
 import retrofit2.Call
 
-class MainPresenter(private val view: MainContract.View, private val interactors: Interactors) :
+class MainPresenter(private val view: MainContract.View, private val retrofitApi: RetrofitApi) :
```
Presenter should not depend on the Api. You need to depend on a Repository, which may be deciding how to deal with the API results (cache if needed etc.)

fsbarata commented on app/src/main/java/com/my/journeyplanner/presenters/main/MainPresenter.kt:
```diff
@@ -18,18 +18,20 @@ class MainPresenter(private val view: MainContract.View, private val interactors
-        interactors.enqueue(call!!)
+        retrofitApi.enqueue(call!!)
+
+        view.showResult(retrofitApi.journeyPlannerResultDomainModel.toString())
```
how did you ensure that enqueue (which is async) has returned before this line (as you're accessing result)

akwc10 committed:
WIP Adding domain model parameter to enqueue callback

Renamed RetrofitApi to JourneyPlannerApi and respective interface.
Simplified Json operation.
Optimised transform operation by replacing add with map.
Reinstated repository at the framework level.
WIP Added empty presenter module.

akwc10 committed:
Domain model custom callback implementation

Replaced cancelAsyncCall() with ICancellable.
Added common.gradle and clean up.
WIP Presenter module.

akwc10 committed:
Gradle update and missing dependency

akwc10 committed:
WIP Decoupling presenter from app module

Moved repository interface back to domain.

akwc10 committed:
Decoupled presenters from app module

Merged getJourneyResults() with enqueue().

akwc10 committed:
Remove state
